src/coreclr/src/gc/gc.cpp

```diff
@@ -1917,11 +1916,19 @@ static void enter_spin_lock (GCSpinLock* spin_lock)
     }
 }
 
-inline BOOL try_enter_spin_lock(GCSpinLock* spin_lock)
+inline
+static BOOL try_enter_spin_lock(GCSpinLock* spin_lock)
 {
     return (Interlocked::CompareExchange(&spin_lock->lock, 0, -1) < 0);
 }
 
+inline
+static void enter_spin_lock (GCSpinLock* spin_lock)
```

Comment:
what's the justification of doing this? also I would not do this with this PR.

Reply:
We take locks very frequently and they are nearly always uncontended. The codepath that handles the contention is very large. It seems wasteful to inline that into callers when the code is not expected to run.

Reply:
I tried to reduce the cost of taking these locks in most common scenarios. These locks are expensive in POH allocations.